### DISCUSSION TITLE
Octoprint Builds

### DISCUSSION THREAD

**Discussion by Millerboy3** [paukstelis/octoprint_deploy#68] [Q&A]
Is there any way to change the builds of octoprint being used? whether it's an RC or a stable, or dev build, or even just a custom repository?

Per instance, or at the prepare stage. I understand I can change it later quite easily. But if it goes based on the template, they will all match correct?

**Comment by paukstelis** [comment]
All instances will use the same build, as they all run out of the same virtual environment. If you change the build version or update OctoPrint in one instance, all instances will be that version (following a restart). If you want to run different version of OctoPrint you need to create different virtual environments.

**Comment by Millerboy3** [comment]
Is there the ability to pull from a custom repository in the build sequence
or changing a flag in the script somewhere?

**Reply by paukstelis** [reply]
No. That's really not something I see many people having an interest in. If you want something custom, just overwrite what is in the /home/$USER/OctoPrint virtual environment.
